adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -2324,6 +2326,19 @@ trait GenomicDataset[T, U <: Product, V <: GenomicDataset[T, U, V]] extends Geno
         convFn.call(v, dsX)
       })
   }
+
+  /**
+   * Filters and replaces the underlying dataset based on overlap with any of a Seq of ReferenceRegions.
+   *
+   * @param querys ReferencesRegions to filter against
+   * @param optPartitionedLookBackNum Optional number of parquet position bins to look back to find start of a
+   *                                  ReferenceRegion, defaults to 1
+   * @return Returns a new DatasetBoundGenomicRDD with ReferenceRegions filter applied.
+   */
+  def filterDatasetByOverlappingRegions(querys: Iterable[ReferenceRegion],
```

Comment:
I'd looove this to move into the new `DatasetBoundGenomicDataset` trait which came in @akmorrow13's https://github.com/bigdatagenomics/adam/commit/60513214660b4f7de9f38c6df9cdd8ede9b25a47, and then this would override `filterByOverlappingRegions`. However, this has a slightly different signature due to the addition of `optPartitionedLookBackNum`. I propose that we either:

* Move this into `DatasetBoundGenomicDataset` and make this method protected, and then override `filterByOverlappingRegions(querys)` to call `filterDatasetByOverlappingRegions(querys)`.
* Or, keep this as is now but open a ticket to close up this API in 0.25.0.

Thoughts?

Unrelated nit: `querys` should be `queries`.

Reply:
Alternatively - upon thinking about it, I don't like this `optPartitionedLookbacknum` as a parameter to the filterByOverlap function anyhow, as this is a config parameter tied to the dataset just like `partitionedBinSize` that should only need be set once when the dataset is created, so we could add it as another optional parameter to the DatasetBoundtype constructor.
 
If we did that, `optPartitionedLookbacknum`,  the parameter could go away making overriding `filterByOverlappingRegions` override in `DatasetBoudGenomicDataset` work cleanly.

Reply:
SGTM!